libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/TracingContextPropagator.java

```diff
@@ -25,6 +27,13 @@ public interface TracingContextPropagator {
      */
     Span extract(Map<String, String> props);
 
+    /**
+     * Extracts current span http header.
+     * @param httpHeader properties
+     * @return current span
+     */
+    Span extract(HttpHeader httpHeader);
```

Comment:
There is definitely no need to build abstractions over `HttpHeader`, the one that uses `Map<String, List<String>> ` should be sufficient:

```suggestion
    Span extract(Map<String, List<String>>  headers);
```

Reply:
Since extract(Map<String, String>  headers); is already there overloading is not allowed.

Reply:
We could use different name, for example `extractFromHeaders`

Reply:
I dont know how could I miss that :)